lib/ansible/modules/database/postgresql/postgresql_facts.py

```diff
@@ -0,0 +1,921 @@
+# ansible standby -m postgresql_facts -a 'filter=repl*'
+
+# Display all facts from databases hosts except settings:
+# ansible databases -m postgresql_facts -a 'filter=!settings'
+
+- name: Collect PostgreSQL version and extensions
+  become: yes
+  become_user: postgres
+  postgresql_facts:
+    filter: ver*,ext*
+
+- name: Collect all facts except settings and roles
+  become: yes
+  become_user: postgres
+  postgresql_facts:
+    filter: "!settings,!roles"
+
+# On FreeBSD with PostgreSQL 9.5 version and lower use pgsql user to become
+# and pass "postgresq" as a database to connect to
+- name: Collect tablespaces and repl_slots facts
+  become: yes
+  become_user: pgsql
+  postgresql_facts:
+    db: postgres
+    filter:
+    - tablesp*
+    - repl_sl*
+
+- name: Collect all facts except settings
+  become: yes
+  become_user: postgres
+  postgresql_facts:
+    filter:
+    - "!settings"
+'''
+
+RETURN = r'''
+version:
+  description: Database server version U(https://www.postgresql.org/support/versioning/).
+  returned: always
+  type: dict
+  sample: { "version": { "major": 10, "minor": 6 } }
+  contains:
+    major:
+      description: Major server version.
+      returned: always
+      type: int
+      sample: 11
+    minor:
+      description: Minor server version.
+      returned: always
+      type: int
+      sample: 1
+databases:
+  description: Information about databases.
+  returned: always
+  type: dict
+  sample:
+  - { "postgres": { "access_priv": "", "collate": "en_US.UTF-8",
+  "ctype": "en_US.UTF-8", "encoding": "UTF8", "owner": "postgres", "size": "7997 kB" } }
+  contains:
+    database_name:
+      description: Database name.
+      returned: always
+      type: dict
+      sample: template0
+      contains:
+        access_priv:
+          description: Database access privileges.
+          returned: always
+          type: str
+          sample: "=c/postgres_npostgres=CTc/postgres"
+        collate:
+          description:
+          - Database collation U(https://www.postgresql.org/docs/current/collation.html).
+          returned: always
+          type: str
+          sample: en_US.UTF-8
+        ctype:
+          description:
+          - Database LC_CTYPE U(https://www.postgresql.org/docs/current/multibyte.html).
+          returned: always
+          type: str
+          sample: en_US.UTF-8
+        encoding:
+          description:
+          - Database encoding U(https://www.postgresql.org/docs/current/multibyte.html).
+          returned: always
+          type: str
+          sample: UTF8
+        owner:
+          description:
+          - Database owner U(https://www.postgresql.org/docs/current/sql-createdatabase.html).
+          returned: always
+          type: str
+          sample: postgres
+        size:
+          description: Database size in bytes.
+          returned: always
+          type: str
+          sample: 8189415
+extensions:
```

Comment:
What I find confusing, is that we have two levels of info crossing here.
Basically, this info is only for the database you are connecting to.
But it is listed on the global level, next to the databases, so it looks as if it where global.

Reply:
I think the nicest way would be to login to all databases and collect this per database.
You could then set this per database on the databases entry level.
And then this entry level should better be info from pg_available_extensions, with the addition of a list of databases it is installed in.

Reply:
Going through all databases is (a little) more intrusive, but it fits to what I expect of a facts module (in that it delivers all facts, not just for one db). You could even switch this on/off, by setting db to '*' or to a (list of) specific database(s), but then '*' should be the default.

Reply:
Good idea